source/legacy_ingress.go

```diff
@@ -0,0 +1,304 @@
+/*
```

Comment:
is this a modified copy of `source/ingress.go`?

Reply:
Oh yeah that's def better, I'm not suggesting if/elsing everything. I think a comment pointing that out would be helpful tho!